### ISSUE TITLE
Assert `RealConstant_t` failed in asr_to_llvm

### ISSUE THREAD

**Issue by Pranavchiku** [lfortran/lfortran#1265]
For
```fortran
program main
    real, dimension(3), parameter :: dpmpar = [epsilon(1.0), &
                                            tiny(1.0), &
                                            huge(1.0)]
end program
```

>LFortran

```console
$ lfortran a.f90 --show-llvm
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/pranavchiku/lfortran/src/bin/lfortran", in _start()
  File "./csu/../csu/libc-start.c", line 392, in __libc_start_main_impl()
  File "./csu/../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main()
  File "/home/pranavchiku/lfortran/src/bin/lfortran.cpp", line 1862, in ??
    return emit_llvm(arg_file, lfortran_pass_manager,
  File "/home/pranavchiku/lfortran/src/bin/lfortran.cpp", line 775, in ??
    = fe.get_llvm(input, lm, pass_manager, diagnostics);
  File "/home/pranavchiku/lfortran/src/lfortran/fortran_evaluator.cpp", line 287, in LCompilers::FortranEvaluator::get_llvm(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&)
    Result<std::unique_ptr<LLVMModule>> res = get_llvm2(code, lm, pass_manager, diagnostics);
  File "/home/pranavchiku/lfortran/src/lfortran/fortran_evaluator.cpp", line 309, in LCompilers::FortranEvaluator::get_llvm2(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&)
    diagnostics, lm.files.back().in_filename);
  File "/home/pranavchiku/lfortran/src/lfortran/fortran_evaluator.cpp", line 342, in LCompilers::FortranEvaluator::get_llvm3(LCompilers::ASR::TranslationUnit_t&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    compiler_options, run_fn, infile);
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 6845, in LCompilers::asr_to_llvm(LCompilers::ASR::TranslationUnit_t&, LCompilers::diag::Diagnostics&, llvm::LLVMContext&, Allocator&, LCompilers::PassManager&, LCompilers::CompilerOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    v.visit_asr((ASR::asr_t&)asr);
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4426, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToLLVMVisitor>::visit_asr(LCompilers::ASR::asr_t const&)
    void visit_asr(const asr_t &b) { visit_asr_t(b, self()); }
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4402, in ??
    case asrType::unit: { v.visit_unit((const unit_t &)x); return; }
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4427, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToLLVMVisitor>::visit_unit(LCompilers::ASR::unit_t const&)
    void visit_unit(const unit_t &b) { visit_unit_t(b, self()); }
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4162, in ??
    case unitType::TranslationUnit: { v.visit_TranslationUnit((const TranslationUnit_t &)x); return; }
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 1284, in LCompilers::ASRToLLVMVisitor::visit_TranslationUnit(LCompilers::ASR::TranslationUnit_t const&)
    visit_symbol(*item.second);
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4429, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToLLVMVisitor>::visit_symbol(LCompilers::ASR::symbol_t const&)
    void visit_symbol(const symbol_t &b) { visit_symbol_t(b, self()); }
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4170, in ??
    case symbolType::Program: { v.visit_Program((const Program_t &)x); return; }
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 2486, in LCompilers::ASRToLLVMVisitor::visit_Program(LCompilers::ASR::Program_t const&)
    declare_vars(x);
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 2967, in void LCompilers::ASRToLLVMVisitor::declare_vars<LCompilers::ASR::Program_t>(LCompilers::ASR::Program_t const&)
    this->visit_expr_wrapper(v->m_symbolic_value, true);
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4421, in LCompilers::ASRToLLVMVisitor::visit_expr_wrapper(LCompilers::ASR::expr_t const*, bool)
    this->visit_expr(*x);
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4489, in LCompilers::ASR::BaseVisitor<LCompilers::ASRToLLVMVisitor>::visit_expr(LCompilers::ASR::expr_t const&)
    void visit_expr(const expr_t &b) { visit_expr_t(b, self()); }
  File "/home/pranavchiku/lfortran/src/libasr/../libasr/asr.h", line 4289, in ??
    case exprType::ArrayConstant: { v.visit_ArrayConstant((const ArrayConstant_t &)x); return; }
  File "/home/pranavchiku/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5227, in LCompilers::ASRToLLVMVisitor::visit_ArrayConstant(LCompilers::ASR::ArrayConstant_t const&)
    ASR::RealConstant_t *cr = ASR::down_cast<ASR::RealConstant_t>(el);
AssertFailed: is_a<T>(*f)
```

**Comment by Pranavchiku** [comment]
This error occurs also in AST -> ASR stage, refer #1161 for this.

**Comment by Pranavchiku** [comment]
Fixed via #1275.
